### DISCUSSION TITLE
Is it possible to override the instrument patch with a JS / Script Setting? (Or Attribute)

### DISCUSSION THREAD

**Discussion by majorcyto** [cifkao/html-midi-player#24] [Q&A] [answered]
I know for CSS purposes we have things like:

```
rect.note[data-instrument="4"]
```

I was wondering if there is anyway to override an instrument patch on a track in the midi file and use a different instrument to play the Track with JS or an attribute selection or something? Then even if the instrument patch is wrong in the midi itself for some reason it could just be 'fixed' by overriding it with the correct instrument ID in the soundfont for Magenta / HTML Midi Player so the user will still hear how it is 'supposed' to sound in the browser preview.

**Comment by cifkao** [comment, marked as answer]
The MIDI files are loaded into [Magenta](https://github.com/magenta/magenta-js/tree/master/music) `NoteSequence` objects (which you can really think of as a JSON representation of the MIDI file). Check out [this tutorial](https://hello-magenta.glitch.me/) to see what they look like. Or you can just run `core.urlToNoteSequence(url)` with the URL of your MIDI file and you will see what the corresponding `NoteSequence` looks like.

So to do what you want, you can use `ns = core.urlToNoteSequence(url)` to load your MIDI file into a `NoteSequence`. Then you can just iterate over all the notes and change their `program`s however you want. Then you do `player.noteSequence = ns` and you're done.

**Reply by majorcyto** [reply]
Sounds.. a bit complicated haha. I was hoping there was just a way to some how select track 1 and then tell it to use midi instrument number 5 within the soundfont within the html-midi-player. But it sounds like that is not a thing haha.

**Reply by cifkao** [reply]
Yeah, no, sorry...
